scripts/SHI/scpSHI_Jetz_complete.R

```diff
@@ -0,0 +1,396 @@
+#Get range map #filter by expert source missing
+source_range_maps <- data.frame(expert_source=expert_source ,
+                                species_name = sp) |>
+  dplyr::mutate(function_name=case_when(
+    expert_source=="IUCN"~ "get_iucn_range_map",
+    expert_source=="MOL"~ "get_mol_range_map",
+    expert_source=="QC" ~ "get_qc_range_map"
+    ),
+    species_path= case_when(
+      expert_source=="IUCN"~ sp,
+      expert_source=="MOL"~ paste0(sp,"_mol"),
+      expert_source=="QC" ~ paste0(sp,"_qc")
+    ))
+
+with(source_range_maps, do.call(function_name,args = list(species_name=species_name)))
+sf_range_map <- st_read(paste0(source_range_maps$species_path,'_range.gpkg'))
+
+print("========== Expert range map successfully loaded ==========")
+
+#get bounding box cropped by country if needed
+if(!is.na(country_code)){
+  ifelse(!is.na(region),
+         sf_area_lim1 <<- gadm(country=country_code, level=1, path=tempdir()) |> st_as_sf() |> st_make_valid() |> filter(NAME_1==region),
+         sf_area_lim1 <<- gadm(country=country_code, level=0, path=tempdir()) |> st_as_sf() |> st_make_valid()
+  )
+  
+  sf_area_lim1_crs <- sf_area_lim1 |> st_transform(sf_crs)
+  
+  sf_area_lim2 <- sf_range_map |> st_make_valid()
+  sf_area_lim2_crs <- sf_area_lim2 |> st_transform(sf_crs)
+  
+  sf_area_lim <<- st_intersection(sf_area_lim2 |> st_transform(st_crs(sf_area_lim1)),sf_area_lim1,dimension="polygon")
+  sf_area_lim_crs <<- st_intersection(sf_area_lim2_crs,sf_area_lim1_crs,dimension="polygon") |> st_make_valid() |> 
+    st_collection_extract("POLYGON")
+  
+  if(!is.null(st_crs(sf_area_lim)$units)){
+    sf_ext <<- st_bbox(sf_area_lim |> st_buffer(spat_res*10))
+  }else{
+    sf::sf_use_s2(FALSE)
+    sf_ext <<- st_bbox(sf_area_lim |> st_buffer(spat_res*0.00001*10))
+  }
+  print(sf_ext)
+  
+  if(!is.null(st_crs(sf_area_lim_crs)$units)){
+    sf_ext_crs <<- st_bbox(sf_area_lim_crs |> st_buffer(spat_res*10))
+  }else{
+    sf::sf_use_s2(FALSE)
+    sf_ext_crs <<- st_bbox(sf_area_lim_crs |> st_buffer(spat_res*0.00001*10))
+  }
+  print(sf_ext_crs)
+
+}else{
+  sf_area_lim <<- sf_range_map
+  sf_area_lim_crs <<- sf_area_lim |> st_transform(sf_crs)
+  
+  if(!is.null(st_crs(sf_area_lim)$units)){
+    sf_ext <<- st_bbox(sf_area_lim |> st_buffer(spat_res*10))
+  }else{
+    sf::sf_use_s2(FALSE)
+    sf_ext <<- st_bbox(sf_area_lim |> st_buffer(spat_res*0.00001*10))
+  }
+  print(sf_ext)
+  
+  if(!is.null(st_crs(sf_area_lim_crs)$units)){
+    sf_ext_crs <<- st_bbox(sf_area_lim_crs |> st_buffer(spat_res*10))
+  }else{
+    sf::sf_use_s2(FALSE)
+    sf_ext_crs <<- st_bbox(sf_area_lim_crs |> st_buffer(spat_res*0.00001*10))
+  }
+  print(sf_ext_crs)
+}
+
+
+#1.4 Final range----------------------------------------------------------------
+#Create raster
+r_frame <- rast(terra::ext(sf_ext_crs),resolution=spat_res)
+crs(r_frame) <- srs_cube
+values(r_frame) <- 1
+r_suitability_map <- terra::mask(r_frame,vect(as(sf_area_lim_crs,"Spatial")))
```

Comment:
This object can create some confusion with SDMs

Reply:
changed to r_range_map